handlers_test.go

```diff
@@ -57,3 +59,58 @@ func TestHandlers(t *testing.T) {
 	assert.Equal(t, h1, hmap.find(m1b))
 	assert.Equal(t, h2, hmap.find(m2b))
 }
+
+func procedure(svc, method string) string {
+	return fmt.Sprintf("%s::%s", svc, method)
+}
+
+func TestUserHandlerWithIgnore(t *testing.T) {
+	const (
+		svc          = "svc"
+		method       = "method"
+		ignoreMethod = "ignoreMethod"
+		runs         = 3
+	)
+
+	ch, err := NewChannel(svc, nil /* ChannelOptions */)
+	require.NoError(t, err, "error creating a TChannel channel")
+
+	userCounter, channelCounter := map[string]int{}, map[string]int{}
+	ch.Register(recorderHandler(channelCounter), ignoreMethod)
+
+	u := userHandlerWithIgnore{
+		ch: ch,
+		ignore: map[string]struct{}{
+			procedure(svc, ignoreMethod): struct{}{},
+		},
+		userHandler: recorderHandler(userCounter),
+	}
+	// need to populate connection.log to avoid nil pointer
+	conn := &Connection{}
+	conn.log = ch.log
+	call := &InboundCall{
+		serviceName:  svc,
+		method:       []byte(method),
+		methodString: method,
+		conn:         conn,
+	}
+	ignoreCall := &InboundCall{
+		serviceName:  svc,
+		method:       []byte(ignoreMethod),
+		methodString: ignoreMethod,
+		conn:         conn,
+	}
```

Comment:
instead of creating these objects ourselves, let's create a real channel with the options set, and validate end-to-end

we generally avoid tests that create partial objects in the tchannel codebase, and use more integration-style tests